.github/workflows/publish_charm.yaml

```diff
@@ -78,9 +159,27 @@ jobs:
           rm -rf .* * || :
           cp -rp $TEMP_DIR/. .
           rm -rf $TEMP_DIR
+      - name: Get charm name
+        id: get-charm-name
+        run: echo "CHARM_NAME=$([ -f metadata.yaml ] && yq '.name' metadata.yaml || echo UNKNOWN)" >> $GITHUB_ENV
```

Comment:
Out of scope for this PR I guess, but I'm wondering if it would be better to fail if the name is not found :D

Reply:
Indeed